src/modules/billing/services/charge-processor-service/lib/agreements.js

```diff
@@ -5,8 +5,10 @@ const helpers = require('@envage/water-abstraction-helpers');
 const dateHelpers = require('./date-helpers');
 const DateRange = require('../../../../../lib/models/date-range');
 
+// filter out agreements if they have been deleted
 const isBillingAgreement = licenceAgreement =>
-  licenceAgreement.agreement.isTwoPartTariff() || licenceAgreement.agreement.isCanalAndRiversTrust();
+  (licenceAgreement.agreement.isTwoPartTariff() && licenceAgreement.dateDeleted === null) ||
```

Comment:
Should this be `licenceAgreement.agreement.dateDeleted` instead of `licenceAgreement.dateDeleted`?

Reply:
I think the licence agreement mapper might need to be changed as well but at the moment the both the licence agreement mapper as well as the agreement mapper maps the date deleted so here just went down the shortest route.